source/base/quadrature_lib.cc

```diff
@@ -1014,6 +1014,116 @@ QWeddle<dim>::QWeddle ()
   Quadrature<dim> (QWeddle<dim-1>(), QWeddle<1>())
 {}
 
+  template <int dim>
+  QTelles<dim>::QTelles (
+      const Quadrature<1> base_quad, const Point<dim> singularity)
+      :
+  /**
+  * We need the explicit implementation if dim == 1. If dim > 1 we use the
+  * former implementation and apply a tensorial product to obtain the higher
+  * dimensions.
+  **/
+      Quadrature<dim>(
+          dim == 2 ?
+          QAnisotropic<dim>(
+              QTelles<1>(base_quad, Point<1>(singularity[0])),
+              QTelles<1>(base_quad, Point<1>(singularity[1]))) :
+          dim == 3 ?
+          QAnisotropic<dim>(
+              QTelles<1>(base_quad, Point<1>(singularity[0])),
+              QTelles<1>(base_quad, Point<1>(singularity[1])),
+              QTelles<1>(base_quad, Point<1>(singularity[2]))) :
+          Quadrature<dim>())
+  {
+  }
+
+  template <int dim>
+  QTelles<dim>::QTelles (
+      const unsigned int n, const Point<dim> singularity)
+      :
+  /**
+  * In this case we map the standard Gauss Legendre formula using the given
+  * singularity point coordinates.
+  **/
+      Quadrature<dim>(QTelles<dim>(QGauss<1>(n), singularity))
+  {}
+
+
+
+  template <>
+  QTelles<1>::QTelles (
+      const Quadrature<1> base_quad, const Point<1> singularity)
+      :
+  /**
+  * We explicitly implement the Telles' variable change if dim == 1.
+  **/
+      Quadrature<1>(base_quad)
+  {
+      /**
+      * We define all the constants to be used in the implementation of
+      * Telles' rule
+      **/
+      double eta_bar = singularity[0] * 2 - 1;
+      double eta_star = eta_bar * eta_bar - 1;
```

Comment:
What about <code>const double eta_bar</code> and <code>const double eta_star</code>?